tests/kohana/Log/FileTest.php

```diff
@@ -168,6 +168,39 @@ public function test_writer_does_not_create_existing_file_but_only_appends(array
 		$this->assertSame($expected_file_contents, $writer->get_written_logs());
 	}
 
+	/**
+	 * Provider for test_created_files_folders_permissions
+	 */
+	public function provider_created_files_folders_permissions()
+	{
+		return [
+			[0777, 0666, TRUE],
+			[0700, 0600, TRUE],
```

Comment:
I think maybe it's better to explicitly assert the mode of the created directory/file after the logs have been written? Possibly also assert that it doesn't modify the attributes of an already-existing directory even if the dir_mode is different?

I think there shouldn't be issues between CI environments/users/etc because the virtual filesystem should behave the same everywhere, it's not tied to the OS environment in any way.